pkgs/by-name/fr/freedownloadmanager/package.nix

```diff
@@ -0,0 +1,116 @@
+  gtk3,
+  atk,
+  cairo,
+  gdk-pixbuf,
+  pango,
+  openssl,
+  icu,
+  mysql80,
+  libdrm,
+  autoPatchelfHook,
+  patchelf,
+  desktop-file-utils,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "freedownloadmanager";
+  version = "6.26.1.6177";
+
+  src = fetchurl {
+    url = "http://debrepo.freedownloadmanager.org/pool/main/f/freedownloadmanager/freedownloadmanager_${finalAttrs.version}_amd64.deb";
+    hash = "sha256-yjyRB/4vnVXB8ZmicZxsrENCCUtp7bRR0YY5x0JtmTg=";
+  };
+
+  nativeBuildInputs = [
+    dpkg
+    autoPatchelfHook
+    patchelf
+    kdePackages.wrapQtAppsHook
+    desktop-file-utils
+  ];
+
+  buildInputs = [
+    kdePackages.qtbase
+    kdePackages.qtdeclarative # QML
+    kdePackages.qtmultimedia
+    kdePackages.qtwayland
+    kdePackages.qt5compat
+
+    # GTK3 dependencies (likely for QPA platform theme like qt6gtk2)
+    gtk3
+    atk
+    cairo
+    gdk-pixbuf
+    pango
+
+    openssl
+    icu
+    mysql80 # Provides libmysqlclient.so for the bundled Qt SQL plugin
+
+    libdrm
+  ];
+
+  preFixup = ''
+    local lib_dir="$out/opt/freedownloadmanager/lib"
+    rm -vf $lib_dir/libQt6*.so.6
+    rm -vf $lib_dir/libicu*.so.*
+    rm -vf $lib_dir/libcrypto.so*
+    rm -vf $lib_dir/libssl.so*
+    rm -vrf $out/opt/freedownloadmanager/qml
+  '';
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/bin $out/opt $out/share/applications $out/share/pixmaps
+
+    cp -r opt/freedownloadmanager $out/opt/
+
+    if [ -d usr/share/applications ]; then
+      cp -r usr/share/applications/* $out/share/applications/
+    else
+      echo "ERROR: usr/share/applications directory not found after unpack!"
+      exit 1
+    fi
+
+    ln -s $out/opt/freedownloadmanager/fdm $out/bin/fdm
+    ln -s $out/opt/freedownloadmanager/icon.png $out/share/pixmaps/freedownloadmanager.png
+
+    local desktopFile="$out/share/applications/freedownloadmanager.desktop"
+    if [ ! -f "$desktopFile" ]; then
+        echo "ERROR: Desktop file not found at $desktopFile after copy!"
+        exit 1
+    fi
+    desktop-file-edit --set-key=Exec --set-value=fdm "$desktopFile"
+    desktop-file-edit --set-key=Icon --set-value=freedownloadmanager "$desktopFile"
+
+    # patch MimerSQL plugin dependency (obscure, safe to remove dep)
+    local sql_plugin_dir="$out/opt/freedownloadmanager/plugins/sqldrivers"
+    if [ -f "$sql_plugin_dir/libqsqlmimer.so" ]; then
+      patchelf --remove-needed libmimerapi.so "$sql_plugin_dir/libqsqlmimer.so" || echo "Warning: Failed to patch libqsqlmimer.so"
+    fi
+
+    # verify MySQL plugin exists, autoPatchelfHook handles its deps now.
+    if [ ! -f "$sql_plugin_dir/libqsqlmysql.so" ]; then
+      echo "Warning: MySQL plugin libqsqlmysql.so not found."
+    fi
+
+    runHook postInstall
+  '';
+
+  meta = {
+    description = "Download manager supporting many protocols";
+    homepage = "https://www.freedownloadmanager.org";
+    license = lib.licenses.unfree;
+    platforms = lib.platforms.linux;
```

Comment:
This is technically incorrect, as only arm64 is supported by FDM debrepo. You should revert it to `[ "x86_64-linux" ] `.

Reply:
Well, even on the regular download on the FDM homepage they explicitly state "64-bit only".